classes/robot/crawler.php

```diff
@@ -38,19 +38,20 @@
 class crawler {
 
     /**
-     * @var the config object
-     */
-    protected $config;
-
-    /**
-     * Robot constructor
+     * Returns configuration object if it has been initialised.
+     * If it is not initialises then it creates and returns it.
+     *
+     * @return mixed hash-like object or defaults false is no config found
      */
-    public function __construct() {
-
-        $this->config = get_config('local_linkchecker_robot');
-        if (!property_exists ($this->config, 'crawlstart') ) {
-            $this->config->crawlstart = 0;
-        }
+    public static function get_config() {
+        $normalisedparams = array(
```

Comment:
This isn't really normalised, this is just the defaults, so I'd call this '$defaults'. The normalized results is what you get after the array merge or anything else you might do to it afterwards